core/src/main/scala/quasar/plugin/postgres/CsvCreateSink.scala

```diff
@@ -104,9 +121,16 @@ object CsvCreateSink extends Logging {
             createTableIfNotExists(log)(tbl, colSpecs)
         }
 
+      ensureTempTable =
+        dropTableIfExists(log)(tempTbl) >> createTable(log)(tempTbl, colSpecs)
+
       copy0 =
-        Stream.eval(ensureTable).void ++
-          doCopy.translate(λ[FunctionK[CopyManagerIO, ConnectionIO]](PHC.pgGetCopyAPI(_)))
+        Stream.eval(verifyExistence).void ++
+          Stream.eval(ensureTempTable).void ++
+          doCopyToTemp.translate(λ[FunctionK[CopyManagerIO, ConnectionIO]](PHC.pgGetCopyAPI(_))) ++
+          Stream.eval(ensureTable).void ++
+          Stream.eval(insertInto(log)(tempTbl, tbl)).void ++
```

Comment:
The important thing is that we won't be locking the actual (i.e. not temporary) table until this line. So, we'll be locking it for less time. We do note that, in the case that the actual table has an index on it, the load will be slower, and this is acceptable. The aim is to lock the actual table for less time than we're locking it currently. Currently we lock it for the duration of the entire load from the source; now we lock it for less time (assuming that the `INSERT INTO` takes less time than entire load as it's scoped currently).

Reply:
I do think, as you posited in your other comment, that rename + drop will be faster than the insert in the replace case. I'm not sure which of drop -> rename or rename -> drop has minimal locking, but either should be much less than now 👍 

We also might consider using a temporary table in the append case, they're much faster to insert into due to their ephemeral nature.

Reply:
I'll look into switching over to drop+rename in the replace case.

This PR introduces using a temporary table in the append case: it copies into a temporary table and then inserts the contents of the temporary table into the actual table. So, not sure I understand your suggestion for the append case?

Reply:
Pushed up a commit to rename in the replace case.

Reply:
@alissapajer Ah, I was referring specifically to the `CREATE TEMPORARY TABLE` feature, apologies for the ambiguity.

Reply:
Should we `CREATE TEMPORARY TABLE` in all the cases, actually? What was the reasoning for just doing this in the append case?

Reply:
Temp tables are deleted automatically at the end of the current session, so we can only use them when they really are temporary.